soda/contracts/design_decisions.md

```diff
@@ -0,0 +1,52 @@
+# Soda data contract YAML format
+
+### No data source & connection details
+
+We leave datasource, connection and schema out of the contract files
+  * Pro
+    * The same contract file can be used to run in CI/CD
+    * Leveraging the existing Soda configuration files
+  * Con:
+    * more CLI args
+    * less repeatable
+    * less sharing of the sharable connection details & profiles
+
+### Column names as keys
+
+* Nicer YAML. A list of columns with a `name` key is less readable.
+* Still JSON Schema enforcable YAML
+
+### Keys without spaces nor variables
+
+No spaces in keys.  No parsing of keys.  No variable parts in keys except for the column names.
+
+* Pro:
+  * More JSON compliant
+  * More validation from JSON schema
+  * More expected and in line with people's expectations
+* Con:
+  * Not similar to SodaCL
```

Comment:
For what it's worth, from a user perspective, having to mentally shift from SodaCL syntax to sort-of-SodaCL-syntax for data contracts could be annoying. If the syntax validation is already a challenge for users (myself included, and as evidenced by the forever absent extra indent or colon), we ought to consider pushing for the same syntax to reduce even this small amount of friction.

Reply:
I think the better JSON schema support and user expectations for YAML in this case win from the consistency with SodaCL.

I also don't think that engineers will switch regularly between Soda Contract YAML and SodaCL YAML.  I would think it's going to be mostly new users that will use the contract language and only that.

Reply:
They won't regularly switch? Even though a user can add SodaCL checks to the same file?